packages/devtools_app/lib/src/timeline/flutter/timeline_screen.dart

```diff
@@ -79,9 +83,23 @@ class TimelineScreenBodyState extends State<TimelineScreenBody>
 
     cancel();
     addAutoDisposeListener(controller.timelineModeNotifier);
-    autoDispose(controller.fullTimeline.onTimelineProcessed.listen((_) {
-      setState(() {});
-    }));
+    addAutoDisposeListener(controller.fullTimeline.recordingNotifier, () {
```

Comment:
this logic is duplicated. maybe add a helper widget that supports using these value notifiers to decide the loading status to show and leaving the other code using ValueListenableBuilder as is?

Reply:
Added a todo to do this in a follow up CL